### ISSUE TITLE
Add capability to rerun failed action on PR

### ISSUE THREAD

**Issue by martin-majlis-s1** [open-telemetry/opentelemetry-collector-contrib#24250]
### Describe the issue you're reporting

It's happening from time to time that action fails on some temporary issue (timeout, connectivity issues). Currently the author of the PR does not have permission to rerun it. It would be great, if this would be possible.

This document describes, how has the permission to do it - https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role

Maybe the action could be triggered by writing some comment?

**Comment by tomaz-s1** [comment]
Agreed, that would be a useful functionality to have, although I'm not 100% sure Github offers this permission for non-collaborators / maintainers.

Right now they way we are "working around" this issue is to simply sync our feature branch with the latest ``main`` branch and that will trigger a new workflow run. 

This is of course not the most ideal approach, but it works relatively fine in this project since it has a lot of activity and new commits will appear in the main branch sooner or later. There are other workarounds possible such as dummy commits, etc., but those usually create noise and are not desired.
